Sources/App/Services/CallRecordingService.swift

```diff
@@ -39,45 +59,79 @@ actor CallRecordingService {
             existingFileNames = []
         }
         logger.info("Found \(existingFileNames.count) existing recordings")
-
+        
         let recordings = try await api.fetchRecordings()
         logger.info("Found \(recordings.count) recordings in Twilio")
         
         for recording in recordings {
-            let fileNameSuffixWAV = "_" + recording.sid + ".wav"
-            let fileNameSuffixJSON = "_" + recording.sid + ".json"
-            
-            if existingFileNames.contains(where: { $0.hasSuffix(fileNameSuffixWAV) }) {
+            if existingFileNames.contains(where: { $0.hasSuffix(recording.sid + ".wav") }) || recording.errorCode != nil {
                 continue
             }
                         
             do {
-                let call = try await api.fetchCall(sid: recording.callSid)
-                guard let twilioDate = parseTwilioDate(from: recording.dateCreated) else {
-                    throw Abort(.unprocessableEntity, reason: "Invalid date format in recording metadata")
-                }
-                let fileNamePrefix = fileName(phoneNumber: call.from, date: twilioDate, internalTestingMode: false)
-                let wavURL = directory.appending(component: fileNamePrefix + fileNameSuffixWAV)
-                let jsonURL = directory.appending(component: fileNamePrefix + fileNameSuffixJSON)
-                let data = try await api.fetchMediaFile(sid: recording.sid)
-                try data.write(to: wavURL)
-                
-                if let encryptionDetails = recording.encryptionDetails {
-                    let jsonData = try JSONEncoder().encode(
-                        CallRecordingMetadata(
-                            cek: encryptionDetails.encryptedCEK,
-                            iv: encryptionDetails.initialVector
-                        )
-                    )
-                    try jsonData.write(to: jsonURL)
-                }
-                logger.info("Successfully downloaded recording file for \(recording.sid) at \(wavURL).")
+                let outputURL = try await storeRecording(recording)
+                logger.info("Successfully downloaded recording file for \(recording.sid) in \(outputURL).")
             } catch {
                 logger.error("Failed to download and store recording file for \(recording.sid): \(error.localizedDescription)")
             }
         }
     }
     
+    private func storeRecording(_ recording: TwilioRecording) async throws -> URL {
+        let call = try await api.fetchCall(sid: recording.callSid)
```

Comment:
Would be good to add some sanity checks here that we only decode completely recorded calls as noted in #47

Reply:
The failed recordings are filtered in the function above by checking out the `errorCode` property - so this function is only really called when the recording has been successful.

Reply:
Nice, thank you 👍